packages/cli/src/lib/snapshot/reportPreviewer/reportPreviewer.ts

```diff
@@ -32,19 +41,65 @@ export class ReportViewer {
       ReportViewer.defaultOperationsToDisplay.concat(operationsToDisplay);
   }
 
-  public display(): void {
+  public async display(): Promise<void> {
     this.printTable();
+    await this.reporter
+      .setReportHandler(
+        SnapshotReportStatus.ERROR,
+        this.getReportErrorsHandler()
+      )
+      .setReportHandler(
+        SnapshotReportStatus.NO_CHANGES,
+        this.getNoChangesReportHandler()
+      )
+      .setReportHandler(
+        SnapshotReportStatus.MISSING_VAULT_ENTRIES,
+        this.getMissingEntriesHandler()
+      )
+      .handleReport();
+  }
 
-    if (!this.reporter.isSuccessReport()) {
-      this.handleReportErrors();
-    }
+  private getMissingEntriesHandler(): (
+    this: SnapshotReporter
+  ) => void | Promise<void> {
+    const missingVaultEntryPrinter = (entries: string[]) =>
+      ReportViewer.printAbridgedMessages(
+        entries,
+        ReportViewer.entryPlurable,
+        yellow
+      );
+    return function (this: SnapshotReporter) {
+      const entries = Array.from(this.missingVaultEntries);
+      CliUx.ux.log(
+        yellow(
+          `Missing vault ${pluralizeIfNeeded(
+            ReportViewer.entryPlurable,
+            entries.length
+          )}:`
+        )
+      );
+      missingVaultEntryPrinter(entries);
+    };
+  }
```

Comment:
TL;DR: Leverage the 'printAbridgedMessage' extracted from the error printer, and feed him the missing entries crunched by the reporter.